### PR TITLE
ci(flask): upgrade tests to support v3 of flask

### PR DESCRIPTION
**Pull request by zarirhamza**
## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: riotfile.py [hunk 1]
```diff
@@ -1053,8 +1050,36 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
                     pkgs={
                         "flask": [
                             "~=2.0.0",
-                            "~=2.2.0",  # latest 2.2
+                            "~=2.3.0",  # latest 2.2
```
Comment (emmettbutler) [inline riotfile.py:1053]: This comment is wrong

**Review by emmettbutler — APPROVED** [review]

File: tests/contrib/flask/test_flask_appsec_iast.py [hunk 2]
```diff
@@ -758,7 +774,11 @@ def test_sqli():
                 _iast_enabled=False,
             )
         ):
-            self.client.set_cookie("localhost", "test-cookie1", "sqlite_master")
+            if werkzeug_version.startswith("3."):
+                self.client.set_cookie(domain="localhost", key="test-cookie1", value="sqlite_master")
```
Comment (mabdinur) [inline tests/contrib/flask/test_flask_appsec_iast.py:778]: Can we use the function call with keyword arguments for both werkzug v2 and v3?

**Review by mabdinur — COMMENTED** [review]
Can we reduce the versions of  flask we test? We could do the min and max supported version for each python version (we should still test every major release but there is no need to test consecutive minor releases)

**Comment by zarirhamza** [comment]
Rewrote the tests so that we test the most recent major version release for each python runtime per major version of flask. No longer testing a range of minor versions within a major version

**Review by mabdinur — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**